src/forest.rs

```diff
@@ -170,14 +199,14 @@ where
 
     pub fn all_splits<'a>(
         &'a self,
-        node: Node<'i, P>,
-    ) -> impl Iterator<Item = (Node<'i, P>, Node<'i, P>)> + Clone + 'a
+        node: Node<'i, G>,
+    ) -> impl Iterator<Item = (Node<'i, G>, Node<'i, G>)> + Clone + 'a
     where
-        P: 'a,
+        G::NodeKind: 'a,
     {
         match self.grammar.node_shape(node.kind) {
             NodeShape::Split(left_kind, right_kind) => self
-                .possible_splits
+                .possibilities
```

Comment:
The structure of this, aside from the lambda in `.map(...)` looks as the one in `all_choices` -- refactor?
There's also some duplication with `one_split`.

Reply:
(I agree, but this pertains to #13)